CMake/PickyWarnings.cmake

```diff
@@ -233,13 +233,23 @@ if(PICKY_COMPILER)
       string(REPLACE "-Wno-" "-W" _ccopt_on "${_ccopt}")
       check_c_compiler_flag(${_ccopt_on} ${_optvarname})
       if(${_optvarname})
-        set(WPICKY "${WPICKY} ${_ccopt}")
+        list(APPEND WPICKY "${_ccopt}")
       endif()
     endforeach()
   endif()
 endif()
 
+# clang-cl
+if(CMAKE_C_COMPILER_ID STREQUAL "Clang" AND MSVC)
+  # list(TRANSFORM WPICKY PREPEND "/clang:") # since CMake 3.12
+  set(_wpicky "")
+  foreach(_ccopt IN LISTS WPICKY)
+    list(APPEND _wpicky "/clang:${_ccopt}")
+  endforeach()
+  set(WPICKY ${_wpicky})
+endif()
+
 if(WPICKY)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WPICKY}")
+  add_compile_options(${WPICKY})
   message(STATUS "Picky compiler options:${WPICKY}")
```

Comment:
This seems to make picky option passed to tests too, which makes them fail.
Also the message shows semicolons.

I wonder if converting to a string and keep using `CMAKE_C_FLAGS` would
fix both?

```cmake
string(REPLACE ";" " " WPICKY "${WPICKY}")
```

Reply:
done

Reply:
> This seems to make picky option passed to tests too, which makes them fail.

What problems do we get in the tests with picky options?

In configure builds we pass on all the picky compiler options to the tests as well without problems.

Reply:
I have to admit it was just a super-quick assesment.

Comparing the two runs it seems the first iteration of this PR got `HAVE_CLOCK_GETTIME_MONOTONIC` and `HAVE_CLOCK_GETTIME_MONOTONIC_RAW` _misdetected_. These are not normally detected for macOS, and detecting them caused some macOS jobs to fail building.

Fail run: https://github.com/curl/curl/actions/runs/11414285596/job/31762998471#step:7:384
Normal run: https://github.com/curl/curl/actions/runs/11418068109/job/31771077457#step:7:384

As it turns out the problem was the reverse of my quick assessment: Before this patch picky warning _were_ passed to tests, and the first iteration of this patch (using `add_compile_options()`) stopped doing that, letting these two tests pass.

Fail test: https://github.com/curl/curl/actions/runs/11414285596/job/31762998471#step:8:5205
Normal test: https://github.com/curl/curl/actions/runs/11418068109/job/31771077457#step:8:5284

TL;DR: we do pass picky warnings to tests and the temp problem was not passing them.

Reply:
(which makes me think we still have a problem for builds with picky warnings explicitly disabled!)